csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,68 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || !this.GetType().Equals(obj.GetType()))
```

Comment:
Checkout the code generated by VS, it is a simpler implementation:

[https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2017](https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2017)

```C#
public override bool Equals(object obj)
{
    return Equals(obj as XXX);
}

public bool Equals(XXX other)
{
    return other != null && ...
}
```

Plus, the way you have it below, `Equals(Schema other)` will null ref if other is null.